app/src/main/java/ar/com/wolox/android/example/ui/viewpager/news/NewsPresenter.kt

```diff
@@ -1,32 +1,66 @@
+class NewsPresenter @Inject constructor(private val postRepository: PostRepository) : CoroutineBasePresenter<NewsView>() {
 
-    fun onInit() {
-        setListNews()
-    }
-
-    private fun setListNews() {
-        var list = ArrayList<ItemNewsModel>()
-        for (num in 1..20) {
-            if (num % 2 == 0) {
-                list.add(ItemNewsModel(R.drawable.ic_profile_icon, "Ali Connors", "I´ll be in you neighborhood doing errands...", "2021-07-29T11:00:00.000Z", R.drawable.ic_favorite))
-            } else {
-                list.add(ItemNewsModel(R.drawable.ic_profile_icon, "Ali Connors", "I´ll be in you neighborhood doing errands...", "2021-07-29T09:00:00.000Z", R.drawable.ic_favorite_border))
-            }
-        }
-        if (list.isEmpty()) {
-            view?.showEmptyNews()
-        } else {
-            view?.showNewsList(list)
+    var list = ArrayList<Page>()
+    var next_page = 0
+    var countNews = 0
+    val offset = 10
+    private var respuesta = ArrayList<Page>()
+    fun onInit() = launch {
+        countNews += 10
+        networkRequest(postRepository.getAllNews()) {
+                onResponseSuccessful { response ->
+                    next_page = response!!.next_page
+                    respuesta = response.page as ArrayList<Page>
+                    if (respuesta != null) {
+                        for (i in 0..countNews) {
+                            list.add(respuesta[i])
+                        }
+                    }
+                    if (list.isEmpty()) {
+                        view?.showEmptyNews()
+                    } else {
+                        view?.showNewsList(list)
+                    }
+                }
+                onResponseFailed { _, _ -> view?.toast("Error en el servidor") }
+                onCallFailure { view?.toast("Error en la conexion") }
         }
     }
     fun onSwipeRefresh() {
         view?.refreshView()
     }
+    fun loadNextNews() = launch {
+        countNews += 10
+        if (countNews < 20) {
+            for (i in countNews - offset..countNews) {
+                list.add(respuesta[i])
+            }
+            view?.showMoreNewsList(list)
+        } else {
+            countNews = 10
+            networkRequest(postRepository.getNextPageNews(next_page)) {
+                onResponseSuccessful { response ->
+                    next_page = response!!.next_page
+                    respuesta = response.page as ArrayList<Page>
+                    if (respuesta != null) {
+                        for (i in 0..countNews) {
+                            list.add(respuesta[i])
+                        }
+                    }
+                    view?.showMoreNewsList(list)
+                }
+                onResponseFailed { _, _ -> view?.toast("Error en el servidor") }
+                onCallFailure { view?.toast("Error en la conexion") }
+            }
+        }
+    }
```

Comment:
I think you can create a method that does this, so you call that method in the onInit and here.